llvm/include/llvm/CodeGen/MachineFunction.h

```diff
@@ -256,7 +256,7 @@ struct LandingPadInfo {
 
 class LLVM_ABI MachineFunction {
   Function &F;
-  const LLVMTargetMachine &Target;
```

Comment:
I'd expect all of the references in the codegen infrastructure to refer to the codegen dependent subclass

Reply:
I don't think there's a need anymore. I'm hoping `CodeGenCommonTMImpl` is treated as just a set of function implementation, and there won't be any hard requirement on subclassing it if you want to use the target independent code generator (CodeGen) library.

In other words, a Target can just ignore everything in `CodeGenCommonTMImpl` and start from scratch, but still use the CodeGen library.